src/jit/rationalize.cpp

```diff
@@ -1555,17 +1555,13 @@ void Rationalizer::RewriteLdObj(GenTreePtr* ppTree, Compiler::fgWalkData* data)
 {    
 #ifdef FEATURE_SIMD
     Compiler* comp = data->compiler;
+    GenTreeLdObj* ldObj = (*ppTree)->AsLdObj();
 
     // Should come here only if featureSIMD is enabled
     noway_assert(comp->featureSIMD);
-
-    GenTreeLdObj* ldObj = (*ppTree)->AsLdObj();
-    assert(ldObj->OperGet() == GT_LDOBJ);
-
-    unsigned simdSize = 0;
-    var_types baseType = comp->getBaseTypeAndSizeOfSIMDType(ldObj->gtClass, &simdSize);
-    assert(baseType != TYP_UNKNOWN);
-    var_types simdType = comp->getSIMDTypeForSize(simdSize);
+    // And we should only call this with a SIMD type.
+    noway_assert(varTypeIsSIMD(ldObj));
+    var_types simdType = ldObj->TypeGet();
 
     assert(ldObj->TypeGet() == TYP_STRUCT || ldObj->TypeGet() == simdType);
```

Comment:
The second condition of || will always be true since simdType = ldobj->TypeGet() on line 1564.  On line 1563 we are noway asserting that ldObj type is a SIMD type and hence it can never be TYP_STRUCT.  Looks like this assert is no longer useful and can be removed.